src/pkg/util/nvidiautils/nvidiautils.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package nvidiautils
+
+import (
+	"bufio"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/singularityware/singularity/src/pkg/sylog"
+)
+
+func nvidiaContainerCli() ([]string, []string, error) {
+	var strArray []string
+	var bindArray []string
+	var soID = regexp.MustCompile(".so")
+
+	// use nvidia-container-cli (if present)
+	command, err := exec.LookPath("nvidia-container-cli")
+	if err != nil {
+		return nil, nil, nil
+	}
+	cmd := exec.Command(command, "list", "--binaries", "--ipcs", "--libraries")
+	out, err := cmd.Output()
+	if err != nil {
+		return nil, nil, nil
+	}
+	for _, line := range strings.Split(string(out), "\n") {
+		val := soID.FindString(line) // this will disallow binaries (non .so files)
+		if val != "" {               // contains .so
+			if line != "" {
+				// extract the filename from the path
+				fileNames := strings.SplitAfter(line, "/")
+				fileName := fileNames[len(fileNames)-1]
+
+				testString := line + ":/.singularity.d/libs/" + fileName
+				bindArray = append(bindArray, testString)
+				strArray = append(strArray, fileName)
+			}
+		} else { // binary executable
+			bindArray = append(bindArray, line)
+			strArray = append(strArray, line)
+		}
+	}
+	return bindArray, strArray, nil
+}
+
+// GetNvidiaBindPath returns a string array consisting of filepaths of nvidia
+// related files to be added to the BindPaths
+func GetNvidiaBindPath(abspath string) ([]string, error) {
+	var strArray []string
+	var bindArray []string
+	var searchArray []string
+	var commentID = regexp.MustCompile(`#`)
+
+	bindArray, strArray, err := nvidiaContainerCli()
+	if err != nil {
+		sylog.Warningf("no nvidia-container-cli returned: %v", err)
```

Comment:
If we're only returning errors which *shouldn't* be ignored, then we should `sylog.Fataf(...)` instead.